packages/salesforcedx-vscode-apex/src/commands/apexActionController.ts

```diff
@@ -39,19 +40,17 @@ export class ApexActionController {
         async progress => {
           // Step 1: Extract Metadata
           progress.report({ message: nls.localize('extract_metadata') });
-          metadata = isClass
-            ? await this.metadataOrchestrator.extractAllMethodsMetadata(sourceUri)
-            : this.metadataOrchestrator.extractMethodMetadata();
+          metadata = await this.metadataOrchestrator.extractMetadata(sourceUri, !isClass);
           if (!metadata) {
             throw new Error(nls.localize('extraction_failed', type));
           }
 
           // Step 3: Generate OpenAPI Document
           progress.report({ message: nls.localize('generate_openapi_document') });
-          const openApiDocument = this.generateOpenAPIDocument(Array.isArray(metadata) ? metadata : [metadata]);
+          const openApiDocument = await this.generateOpenAPIDocument(metadata);
 
           // Step 4: Write OpenAPI Document to File
-          name = Array.isArray(metadata) ? metadata[0].className : metadata.name;
+          name = isClass ? path.basename(metadata.resourceUri, '.cls') : metadata?.symbols?.[0]?.docSymbol?.name;
           const openApiFileName = `${name}_openapi.yml`;
```

Comment:
Name may be undefined, does undefined mean something is wrong?

Reply:
at that point I don't think so? We'd have already generated the doc and done all the validations, so I don't know if it's that important